libraries/AP_HAL_Linux/Util.cpp

```diff
@@ -94,6 +96,11 @@ void Util::_toneAlarm_timer_tick() {
 void Util::set_system_clock(uint64_t time_utc_usec)
 {
 #if CONFIG_HAL_BOARD_SUBTYPE != HAL_BOARD_SUBTYPE_LINUX_NONE
+    if (system_time_was_set()) {
+        return;
+    }
```

Comment:
ok, now that the checks are in other places you can remove this check. Let this decision be done by the caller and not by the hal implementation here.

Reply:
Leaving it in the hal seems like the right choice to me. The checks were always in vehicle, they are redundant now, and could be removed, but I don't think that this is the desirable action. I don't see a problem with one extra 'if' to be sure, and to make it easier for a human to know what is going on with the code.

Reply:
From my POV we are spreading it on different HALs in a not obvious way, so I don't agree. The checks are redundant now because each vehicle handles it. It doesn't need to be that way, though. It could be done my common vehicle code.